src/docker/job_control.py

```diff
@@ -1,132 +1,245 @@
 #!/usr/bin/env python3
-
-from os import listdir
-import os
-import sys
-import shutil
-from os import path
-import datetime, time, json, urllib
-import boto3
-
-qtimeout=300
-AWS_REGION='us-west-2'
-max_tries=60
-retry_time=15
-
-path = '/dev/shm/test/'
-bucket=os.getenv("OUTPUT_BUCKET")
-bucket='apbs-test-output'
-queue=os.getenv("JOB_QUEUE_NAME")
-queue='apbs-test-job-q'
-
-
-def get_items(sqs,qurl):
-  loop=0
-
-  items = sqs.receive_message( QueueUrl=qurl, MaxNumberOfMessages=1, VisibilityTimeout=qtimeout)
-  
-  while 'Messages' not in items:
-    loop+=1
-    if loop==max_tries:
-      return 0
-    print ("Waiting ....")
-    time.sleep(retry_time)
-    items = sqs.receive_message( QueueUrl=qurl, MaxNumberOfMessages=1, VisibilityTimeout=qtimeout)
-  
-  return items
-
-def update_state(s3,jobid,jobtype,status,output):
-  ups3=boto3.resource('s3')
-  objectfile=jobid+'/'+jobtype+'-status.json'
-  s3obj = ups3.Object(bucket, objectfile)
-  statobj:dict = json.loads(s3obj.get()['Body'].read().decode('utf-8')) 
-  
-  statobj[jobtype]['status']=status
-  statobj[jobtype]['endTime']=time.time()
-  ## FIX
-  statobj[jobtype]['outputFiles']=output
-  
-  object_response:dict = s3.put_object(
-                              Body=json.dumps(statobj),
-                              Bucket=bucket,
-                              Key=objectfile
-  )
-
-def run_code(job,s3):
-  out=1
-  try:
-    job_info:dict = json.loads(job) 
-    if 'job_id' not in job_info:
-      return 1
-  except:
-    print ("Not a json q item")
+"""Software to run apbs and pdb2pqr jobs."""
+
+from os import chdir, getenv, listdir, makedirs, system
+from datetime import datetime
+from enum import Enum
+from json import dumps, loads
+from logging import getLogger
+from time import sleep, time
+from shutil import rmtree
+from typing import Any, Dict, List
+from urllib import request
+from boto3 import client, resource
+
+_LOGGER = getLogger(__name__)
+Q_TIMEOUT = 300
+AWS_REGION = "us-west-2"
+MAX_TRIES = 60
+RETRY_TIME = 15
+
+MEM_PATH = "/dev/shm/test/"
+S3_BUCKET = getenv("OUTPUT_BUCKET")
+QUEUE = getenv("JOB_QUEUE_NAME")
+
+
+class JOBTYPE(Enum):
```

Comment:
the rest of the code/web site uses lower case for these types.  
And i believe that's for the JOBSTATUS too.   The web is exepcted lower case strings.

Reply:
at a minimum I just did lc on these 2 to get the backend to work
And also lc on the runjob check for these names.

Reply:
I think we fixed this.